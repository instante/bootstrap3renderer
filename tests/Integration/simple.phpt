<?php

namespace InstanteTests\Bootstrap3Renderer;

use Instante\Bootstrap3Renderer\BootstrapRenderer;
use Instante\Tests\TestBootstrap;
use Nette\Forms\Form;
use Tester\Assert;

require_once __DIR__ . '/../bootstrap.php';

$form = new Form;
$form->setRenderer(new BootstrapRenderer);
$form->addText('name', 'Your name')->addError('Too short');
$form->addCheckbox('agree', 'I agree');
$form->addImage('picture', '//avatars1.githubusercontent.com/u/13833444?v=3&s=200');
$form->addSelect('sure', 'Are you sure?', ['y' => 'yes', 'n' => 'no']);
$form->addCheckboxList('list', 'List of options', ['a' => 'First', 'b' => 'Second']);
$form->addPassword('password', 'Password')->setRequired();
$form->addButton('btn', 'Push me');
$form->addHidden('you', 'dont see me');
$form->addRadioList('sex', 'Gender', ['m' => 'Male', 'f', 'Female'])->setValue('m')->setDisabled();
$form
    ->addTextArea('text', 'Description')
    ->setValue('Some text')
    ->setOption('description', 'Better be long');
;
$form->addUpload('photo', 'Your photo here');
$form->addSubmit('sub', 'Send me now');
$form->addSubmit('notme', 'But not me')->setDisabled();

ob_start();
require __DIR__ . '/bootstrap.phtml';
$content = ob_get_clean();
file_put_contents(TestBootstrap::$tempDir . '/' . basename(__FILE__) . '.html', simplifyHtmlWhitespaces($content));

ob_start();
$form->render();
$content = ob_get_clean();

Assert::matchFile(__DIR__ . '/expected/simple.htm', simplifyHtmlWhitespaces($content));
