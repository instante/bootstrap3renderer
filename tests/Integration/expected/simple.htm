<form
action=""
method="post"
enctype="multipart/form-data">
<div
class="form-group
has-error">
<label
for="frm-name">Your
name</label>
<input
type="text"
name="name"
id="frm-name"
class="form-control">
<span
class="help-block
text-danger">Too
short</span>
</div>
<div
class="form-group">
<div
class="checkbox">
<label
for="frm-agree">
<input
type="checkbox"
name="agree"
id="frm-agree">I
agree</label>
</div>
</div>
<input
type="image"
name="picture[]"
src="//avatars1.githubusercontent.com/u/13833444?v=3&amp;s=200"
class="btn
btn-primary">
<div
class="form-group">
<label
for="frm-sure">Are
you
sure?</label>
<select
name="sure"
id="frm-sure"
class="form-control">
<option
value="y">yes</option>
<option
value="n">no</option>
</select>
</div>
<div
class="form-group">
<label>List
of
options</label>
<div
class="checkbox">
<label
for="frm-list-a">
<input
type="checkbox"
name="list[]"
id="frm-list-a"
value="a">First</label>
</div>
<div
class="checkbox">
<label
for="frm-list-b">
<input
type="checkbox"
name="list[]"
id="frm-list-b"
value="b">Second</label>
</div>
</div>
<div
class="form-group
required">
<label
for="frm-password">Password</label>
<input
type="password"
name="password"
id="frm-password"
required
data-nette-rules='[{"op":":filled","msg":"This
field
is
required."}]'
class="form-control">
</div>
<input
type="button"
name="btn"
value="Push
me"
class="btn
btn-default">
<div
class="form-group">
<label>Gender</label>
<div
class="radio">
<label
for="frm-sex-m">
<input
type="radio"
name="sex"
id="frm-sex-m"
disabled
value="m">Male</label>
</div>
<div
class="radio">
<label
for="frm-sex-0">
<input
type="radio"
name="sex"
id="frm-sex-0"
disabled
value="0">f</label>
</div>
<div
class="radio">
<label
for="frm-sex-1">
<input
type="radio"
name="sex"
id="frm-sex-1"
disabled
value="1">Female</label>
</div>
</div>
<div
class="form-group">
<label
for="frm-text">Description</label>
<textarea
name="text"
id="frm-text"
aria-describedby="describe-frm-text"
class="form-control">Some
text</textarea>
<span
class="help-block"
id="describe-frm-text">Better
be
long</span>
</div>
<div
class="form-group">
<label
for="frm-photo">Your
photo
here</label>
<input
type="file"
name="photo"
id="frm-photo">
</div>
<input
type="submit"
name="sub"
value="Send
me
now"
class="btn
btn-primary">
<input
type="submit"
name="notme"
disabled
value="But
not
me"
class="btn
btn-primary">
<input
type="hidden"
name="you"
value="dont
see
me">
</form>
